src/libos/src/vm/vm_manager.rs

```diff
@@ -494,6 +534,7 @@ pub struct InternalVMManager {
     chunks: BTreeSet<ChunkRef>, // track in-use chunks, use B-Tree for better performance and simplicity (compared with red-black tree)
     fast_default_chunks: Vec<ChunkRef>, // empty default chunks
     free_manager: VMFreeSpaceManager,
+    gaps: Vec<ChunkRef>, // Memory gaps that shouldn't be accessed by users
```

Comment:
Thank you for your detailed suggestion. After re-consideration, I would insist on keeping the `gaps` field. Firstly because it helps for further extension in the future if we need more memory holes. Secondly, keeping this field helps to verify all the memory is returned when the libOS exits.

Reply:
Removed. Thank you for your review.